ouroboros-consensus-byron/src/Ouroboros/Consensus/Byron/Ledger/Ledger.hs

```diff
@@ -255,26 +255,28 @@ instance LedgerSupportsProtocol ByronBlock where
                 Origin -> SlotNo $ 2 * k
                 At s   -> SlotNo $ unSlotNo s + 1 + (2 * k)
 
-byronEraParams :: Gen.Config -> HardFork.EraParams
-byronEraParams genesis = HardFork.EraParams {
+byronEraParams :: HardFork.SafeBeforeEpoch -> Gen.Config -> HardFork.EraParams
+byronEraParams safeBeforeEpoch genesis = HardFork.EraParams {
       eraEpochSize  = fromByronEpochSlots $ Gen.configEpochSlots genesis
     , eraSlotLength = fromByronSlotLength $ genesisSlotLength genesis
     , eraSafeZone   = HardFork.SafeZone {
           safeFromTip     = 2 * k
-
-          -- @180@ is merely a lower bound on the 'EpochNo' in which the
-          -- Byron-to-Shelley transition could take place. We can update
-          -- it over time, and set it to the precise value once the
-          -- transition has actually taken place.
-        , safeBeforeEpoch = HardFork.LowerBound (EpochNo 180)
+        , safeBeforeEpoch = safeBeforeEpoch
         }
     }
   where
     SecurityParam k = genesisSecurityParam genesis
 
 instance HasHardForkHistory ByronBlock where
   type HardForkIndices ByronBlock = '[ByronBlock]
-  hardForkSummary = neverForksHardForkSummary byronEraParams
+  hardForkSummary =
+      neverForksHardForkSummary (byronEraParams safeBeforeEpoch)
+    where
+      -- @180@ is merely a lower bound on the 'EpochNo' in which the
+      -- Byron-to-Shelley transition could take place. We can update it over
+      -- time, and set it to the precise value once the transition has actually
+      -- taken place.
+      safeBeforeEpoch = HardFork.LowerBound (EpochNo 180)
```

Comment:
I don't think it makes sense to define this bound here, because it will only be used for a standalone Byron chain, which never forks (hence `neverForksHardForkSummary`), so here you can use `UnsafeUnbounded`. Note: this `hardForkSummary` is not compositional, so this instance is not used by the Byron-to-Shelley HFC.

Note to myself: we should do the same thing for Shelley, for any single era block in fact.

Reply:
Done!